### ISSUE TITLE
On Big Endian machines like AIX, we are getting different Hash Key by the MurmurHash3::x86_32()

### ISSUE THREAD

**Issue by tvkai** [microsoft/onnxruntime#12504]
**Describe the bug**
Basically when i try to compile and run this code on AIX , hitting many testcase failures. 

**Urgency**
If there are particular important use cases blocked by this or strict project-related timelines, please share more information and dates. If there are no hard deadlines, please specify none. 

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): AIX
- ONNX Runtime installed from (source or binary): Source
- ONNX Runtime version: latest
- Python version: NA
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source): 10.3.0
- CUDA/cuDNN version:
- GPU model and memory:

**To Reproduce**
- Describe steps/code to reproduce the behavior.
- Attach the ONNX model to the issue (where applicable) to expedite investigation.
-
This is shown up only on AIX machine, ( i guess should have been seen on other big endian platforms also . )
**Expected behavior**

On both ( Little and Big ) endian architectures, it should get the same Hash Key.

**Screenshots**

NA

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.   NA

Basically the issue is .. MurmurHash3::x86_32() internally invokes the getblock32 , getblock which needs fix for big endian platforms

**Comment by skottmckay** [comment]
Would copying this [change](https://github.com/explosion/murmurhash/pull/27/files) work? That's in a completely separate repo but appears to involve the same original murmurhash3 source that we are using?

https://github.com/explosion/murmurhash/blob/c9210577de709cc000894c726101a4750def5022/murmurhash/MurmurHash3.cpp#L66-L94

FYI @edgchen1

**Comment by tvkai** [comment]
yes, that's correct, it should work.

**Comment by skottmckay** [comment]
That's possibly gcc specific though so we probably need a more general purpose compile time solution that is hopefully simpler than https://gist.github.com/jtbr/7a43e6281e6cca353b33ee501421860c#file-endianness-h-L31-L65

**Comment by tvkai** [comment]
As in this file include/onnxruntime/core/framework/endian.h, only platform where macro __BYTE_ORDER__ is not available is WIN32, so having a check as in https://github.com/explosion/murmurhash/blob/c9210577de709cc000894c726101a4750def5022/murmurhash/MurmurHash3.cpp#L66-L94 should work.

**Comment by skottmckay** [comment]
@tvkai As you're building from source would you be able to test out the change in #12549 to validate it works? I don't have a big-endian machine to do able to do so.

**Comment by tvkai** [comment]
@skottmckay Change #12549 is a runtime check and should work for even any future big endian systems that may get added. Fix works on AIX.

**Comment by skottmckay** [comment]
The constexpr should mean only one branch is possible at compile time and the dead branch is easily thrown away, so I don't expect there to be any runtime overhead of an endian-ness check.
